zulip/integrations/clickup/zulip_clickup.py

```diff
@@ -0,0 +1,372 @@
+#!/usr/bin/env python3  # noqa: EXE001
+#
+# A ClickUp integration script for Zulip.
+
+import argparse
+import json
+import os
+import re
+import sys
+import time
+import webbrowser
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+from urllib.error import HTTPError
+from urllib.parse import parse_qs, urlencode, urljoin, urlparse
+from urllib.request import Request, urlopen
+
+EVENT_CHOICES: Dict[str, Tuple[str, ...]] = {
+    "1": ("taskCreated", "taskUpdated", "taskDeleted"),
+    "2": ("listCreated", "listUpdated", "listDeleted"),
+    "3": ("folderCreated", "folderUpdated", "folderDeleted"),
+    "4": ("spaceCreated", "spaceUpdated", "spaceDeleted"),
+    "5": ("goalCreated", "goalUpdated", "goalDeleted"),
+}
+
+
+def clear_terminal_and_sleep(sleep_duration: int = 3) -> Callable[[Any], Callable[..., Any]]:
+    """
+    Decorator to clear the terminal and sleep for a specified duration
+    before and after the execution of the decorated function.
+    """
+    cmd = "cls" if os.name == "nt" else "clear"
+
+    def decorator(func: Any) -> Any:
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            os.system(cmd)  # noqa: S605
+            result = func(*args, **kwargs)
+            time.sleep(sleep_duration)
+            os.system(cmd)  # noqa: S605
+            return result
+
+        return wrapper
+
+    return decorator
+
+
+def process_url(input_url: str, base_url: str) -> str:
+    """
+    Validates that the URL is the same the users zulip app URL.
+    Returns the authorization code from the URL query
+    """
+    parsed_input_url = urlparse(input_url)
+    parsed_base_url = urlparse(base_url)
+
+    same_domain: bool = parsed_input_url.netloc == parsed_base_url.netloc
+    auth_code = parse_qs(parsed_input_url.query).get("code")
+
+    if same_domain and auth_code:
+        return auth_code[0]
+    else:
+        print("Unable to fetch the auth code. exiting")
+        sys.exit(1)
+
+
+class ClickUpAPIHandler:
+    def __init__(
+        self,
+        client_id: str,
+        client_secret: str,
+        team_id: str,
+    ) -> None:
+        self.client_id: str = client_id
+        self.client_secret: str = client_secret
+        self.team_id: str = team_id
+        self.API_KEY: Optional[str] = None
+
+    def make_clickup_request(
+        self, path: str, query: Dict[str, Union[str, List[str]]], method: str
+    ) -> Optional[Dict[str, Any]]:
+        base_url = "https://api.clickup.com/api/v2/"
+        api_endpoint = urljoin(base_url, path)
+
+        if path == "oath/token":
```

Comment:
Hmm, I think I made some changes and forgot to update the PR because it's already "oauth" in my local. Or does it work for you too??

Reply:
It might be working because the API might not be that strict in terms of what headers it is getting (because those are being modified in the if-else block).